src/ui/util/zcl_dbbr_cds_selection_util.clas.abap

```diff
@@ -581,4 +596,78 @@ CLASS zcl_dbbr_cds_selection_util IMPLEMENTATION.
 
   ENDMETHOD.
 
+  METHOD calculate_virtual_elements.
+
+    DATA: lo_calculated_by TYPE REF TO object,
+          lr_t_calc_data   TYPE REF TO data.
+
+    FIELD-SYMBOLS: <lt_selected_data>   TYPE STANDARD TABLE,
+                   <lt_calculated_data> TYPE STANDARD TABLE.
+
+    DATA(lt_virtual_elems) = mo_cds_view->get_annotations(
+      it_annotation_name = VALUE #( ( sign = 'I' option = 'EQ' low = 'OBJECTMODEL.VIRTUALELEMENTCALCULATEDBY' ) ) ).
+
+    CHECK lt_virtual_elems IS NOT INITIAL.
+
+    ASSIGN mr_t_data->* TO <lt_selected_data>.
+
+    LOOP AT lt_virtual_elems ASSIGNING FIELD-SYMBOL(<ls_virtual_elem>).
+
+      TRY.
+          CREATE OBJECT lo_calculated_by TYPE (<ls_virtual_elem>-value+5).
+
+          create_virtual_elem_table_type(
+            EXPORTING iv_virtual_elem = <ls_virtual_elem>-fieldname
+            IMPORTING eo_table_type = DATA(lo_table_type) ).
+
+          CREATE DATA lr_t_calc_data TYPE HANDLE lo_table_type.
+          ASSIGN lr_t_calc_data->* TO <lt_calculated_data>.
```

Comment:
Wouldn't this automatically override the calculated data from the previous iteration?